lib/commands/init.js

```diff
@@ -133,8 +134,16 @@ class Init extends BaseCommand {
     const scriptShell = this.npm.config.get('script-shell') || undefined
     const yes = this.npm.config.get('yes')
 
+    // only send the init-private flag if it is set explicitly to true
+    const opts = { ...flatOptions }
+    if (this.npm.config.get('init-private') === true) {
```

Comment:
```suggestion
    if (!this.npm.config.isDefault('init-private')) {
```

Folks should still be able to have this explicitly set in their package.json, even `false`